File: codes/Garnetwzy/240.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if(matrix.size() == 0 || matrix[0].size() == 0)
            return false;
        int row = 0;
        int colomn = matrix[0].size()-1;
        while(row < matrix.size() && colomn >= 0) {
            if(matrix[row][colomn] == target)
                return true;
            if(matrix[row][colomn] > target) {
                colomn--;
            }else{
                row++;
            }
        }
        return false;
    }
};

File: codes/HellDream/240.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if(matrix.length==0||matrix[0].length==0) return false;
        for(int i=0;i<matrix.length;i++){
            if(search(matrix[i],target)){
                return true;
            }
        }
        return false;
    }
    
    public boolean search(int[] nums, int target){
        int left = 0, right = nums.length-1;
        while(left+1<right){
            int mid = left+(right-left)/2;
            if(nums[mid]==target) return true;
            else if(nums[mid]<target) left = mid;
            else right = mid;
        }
        return nums[left]==target||nums[right]==target;
    }
}